sonic-thermalctld/scripts/thermalctld

```diff
@@ -462,6 +463,15 @@ class TemperatureUpdater(logger.Logger):
         self.temperature_status_dict = {}
         state_db = daemon_base.db_connect("STATE_DB")
         self.table = swsscommon.Table(state_db, TemperatureUpdater.TEMPER_INFO_TABLE_NAME)
+        self.chassis_table = None
+
+        self.is_chassis_system = chassis.is_modular_chassis()
```

Comment:
chassis.is_modular_chassis() looks like a new platform API, does it raise exception?

Reply:
It is a new API. However, in chassis_base.py the default implementation will return False. So, chassis vendors will have explicitly override this method.